test/functional/discovered_hosts_controller_test.rb

```diff
@@ -34,23 +34,23 @@ def test_index_with_custom_column
                        :category => 'Setting::Discovered')
     facts = @facts.merge({"bios_vendor" => "QEMU"})
     discover_host_from_facts(facts)
-    get :index, {}, set_session_user_default_reader
+    get :index, params: {}, session: set_session_user_default_reader
     assert_select "td", /QEMU/
     assert_response :success
   end
 
   def test_show_multiple_actions
     host = discover_host_from_facts(@facts)
-    [:multiple_destroy, :select_multiple_organization, :select_multiple_location].each do |path|
-      xhr :post, path, {:host_ids => [host.id]}, set_session_user
+    [:multiple_destroy, :select_multiple_organization, :select_multiple_location].each do |action|
+      process action, method: :post, params: {:host_ids => [host.id]}, session: set_session_user
```

Comment:
I guess this needs a `, xhr: true` added

Reply:
Added, tests passed without it, was not sure.